src/routes.js

```diff
@@ -494,7 +494,28 @@ const routes = [
       }
     }
   }),
-
+  asyncRoute({
+    path: '/projects/:projectId([1-9]\\d*)/datasets/:datasetName',
+    component: 'DatasetShow',
+    props: true,
+    loading: 'page',
+    key: ({ projectId, datasetName }) =>
+      `/projects/${projectId}/datasets/${encodeURIComponent(datasetName)}`,
+    children: [
+      asyncRoute({
+        path: '',
+        component: 'DatasetOverview',
+        props: true,
+        loading: 'tab',
+        meta: {
+          title: () => [dataset.name],
+          validateData: {
+            project: () => project.permits('dataset.list')
```

Comment:
It'd be great to add a test of this in test/router.spec.js. A Data Collector should be redirected if they try to navigate to this route.